File: C99/백준/Silver/1929. 소수 구하기/소수 구하기.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main() {
    int m, n;
    scanf("%d %d", &m, &n);

    char *isPrime = (char*)malloc(n + 1);
    for (int i = 0; i <= n; i++) isPrime[i] = 1;
    if (n >= 0) isPrime[0] = 0;
    if (n >= 1) isPrime[1] = 0;

    int limit = (int)sqrt((double)n);
    for (int i = 2; i <= limit; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= n; j += i) {
                isPrime[j] = 0;
            }
        }
    }

    for (int x = (m < 2 ? 2 : m); x <= n; x++) {
        if (isPrime[x]) printf("%d\n", x);
    }

    free(isPrime);
    return 0;
}

File: Java/백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int m = Integer.parseInt(st.nextToken());
        int n = Integer.parseInt(st.nextToken());

        boolean[] isPrime = new boolean[n + 1];
        Arrays.fill(isPrime, true);
        if (n >= 0) isPrime[0] = false;
        if (n >= 1) isPrime[1] = false;

        int limit = (int)Math.sqrt(n);
        for (int i = 2; i <= limit; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        StringBuilder sb = new StringBuilder();
        for (int x = Math.max(2, m); x <= n; x++) {
            if (isPrime[x]) sb.append(x).append('\n');
        }
        System.out.print(sb.toString());
    }
}
